### DISCUSSION TITLE
How to validate you’re running a genuine install?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2570] [Ideas / Feature Requests]
I have an idea of selling ready-made Raspberry PIs containing BTCPay Server (and donating proceeds), but want to empower customers to "don't trust, verify" their install. 

Is there a way to make this a reality? One could generate a hash of the install and verify against published versions.

This feature could be added to all installs, giving everyone this option. 

Pitfalls I see are:
- Is this actually a good idea?
- Hashing all essential files and excluding data. Which files?
- What to compare with?
- Comparison will be against an online resource, which can be intercepted and faked. Is HTTPS safe enough for this?
- A factory reset link in the backoffice would be handy to make 100% sure you download the blockchain yourself. 

Any other things I'm missing?

**Comment by Kixunil** [comment]
Just sign the whole RPi SD card image, anyone can then do `gpg -v`. But that is probably too much security if `btcpayserver-docker` is being used since that one doesn't verify everything anyway. To get complete verification you'd have to use [Cryptoanarchy Debian Repository](https://github.com/debian-cryptoanarchy/cryptoanarchy-deb-repo-builder) or [nix-bitcoin](https://github.com/fort-nix/nix-bitcoin/).
